utils/bin/poller/src/main.rs

```diff
@@ -0,0 +1,301 @@
+                    callData: LegacyConstantProduct::commitmentCall { amm }
+                        .abi_encode()
+                        .into(),
+                }),
+            )
+        }
+        false => match (numerator, denominator) {
+            (Ok(n), Ok(d)) => (U256::from_str(&n)?, U256::from_str(&d)?, None),
+            _ => {
+                return Err(eyre::eyre!(
+                    "NUMERATOR and DENOMINATOR env vars must be set for non-legacy AMMs"
+                ))
+            }
+        },
+    };
+
+    // Fifth, we now have relative prices, so we can use the helper to get the order
+    let hint = helper
+        .order(amm, vec![numerator, denominator])
+        .state(overrides)
+        .call_raw()
+        .await
+        .map_or_else(Err, |d| {
+            let dReturn {
+                order,
+                preInteractions,
+                postInteractions,
+                sig,
+            } = IConstantProductHelper::dCall::abi_decode_returns(&d, true).unwrap();
+            Ok((order, preInteractions, postInteractions, sig))
+        });
+
+    // Sixth, use the hint and verify that it can be settled on-chain. This is done by:
+    // 1. Using `simulateDelegateCall` from the settlement contract to the `Multicall3` as we need to call two functions
+    // 2. The first function is the interaction, which is the `commit` function on the AMM
+    // 3. The second function is the `isValidSignature` function on the AMM that would otherwise be called
+    //    normally during the course of the settlement process
+    // 4. We will then verify that the `isValidSignature` function returns valid
+
+    match hint {
+        Ok((order, pre_interactions, post_interactions, sig)) => {
+            println!("\nHint received!");
+            println!("Order: {:?}", order);
+            println!("Pre Interactions: {:?}", pre_interactions.clone());
+            println!("Post Interactions: {:?}", post_interactions.clone());
+            println!("Sig: {:?}", sig);
+
+            let offchain_order = Order::try_from(order).unwrap();
+
+            let domain = eip712_domain! {
+                name: "Gnosis Protocol",
+                version: "v2",
+                chain_id: chain_id,
+                verifying_contract: SETTLEMENT.parse()?,
+            };
+
+            let signing_message = offchain_order.eip712_signing_hash(&domain);
+
+            // To do this, we will make use of Multicall3
+            let payload = IMulticall3::tryAggregateCall {
+                requireSuccess: true,
+                calls: vec![
+                    // Inline mapping of pre_interactions to Multicall3::Call
+                    pre_interactions
+                        .iter()
+                        .map(|interaction| IMulticall3::Call {
+                            target: interaction.target,
+                            callData: interaction.callData.clone(),
+                        })
```

Comment:
Here we should use a function that transforms an interaction into a `IMulticall::Call` that *returns error if value is nonzero*. Otherwise it would be hard to debug what the issue is. Same for the map below.

Reply:
Done.